### ISSUE TITLE
geometry setting in geany.conf has no effect under Ubuntu 16.04, 64 bit, plain Openbox

### ISSUE THREAD

**Issue by Lew-Rockwell-Fan** [geany/geany#1401]
The geometry line in geany.conf has no effect for me under a minimalistic Ubuntu 16.04, 64 bit, built from the mini.iso up with X started with startx, using plain Openbox as sole DE, if you want to call it that. Nearest thing to a work-around I've come up with is to set it to reopen at the previous geometry. THAT works, but it isn't really the behavior I'd prefer. I'd prefer to set a default geometry and always open at that. If I don't set it to reopen at the previous geometry, it opens at about half the dimensions of my monitor horizontally and vertically. Possibly I could set it to open maximized, but that really isn't what I want either. Here is the present line:
`geometry=0;0;1358;713;0;`
No matter how I change the numbers, they have no effect.

**Comment by elextr** [comment]
You have to edit the geany.conf file in some other editor with Geany closed because that setting is overridden with the current position and size when Geany closes.

**Comment by elextr** [comment]
Also note that Blackbox is a stacking window manager, that is it wants to control where windows open, so it may ignore any polite requests Geany makes to return to its saved size and position.

**Comment by codebrainz** [comment]
Openbox != Blackbox && Stacking == Traditional

But yeah, I'd wager it is the WM taking over placement of the window though, there might even be some kind of [placement settings](http://openbox.org/wiki/Help:Configuration#Placement) for it.

**Comment by elextr** [comment]
Oh, **OPEN**box, my bad, misread :)

Yeah, as I said all Geany can do is politely ask the WM, it can totally ignore that request.

**Comment by codebrainz** [comment]
Yeah, for example, from the [`gtk_window_move()` docs](https://developer.gnome.org/gtk3/stable/GtkWindow.html#gtk-window-move):

> Asks the window manager to move window to the given position. Window managers are free to ignore this; most window managers ignore requests for initial window positions (instead using a user-defined placement algorithm) and honor requests after the window has already been shown.

**Comment by Lew-Rockwell-Fan** [comment]
Thank you, both. Mea culpa for tardiness in getting back.  Mea culpa also for not saying I have turned saving window geometry off. Anyway I tried setting geany's size by hand dragging on the edges (which works fine FOR THAT INSTANCE), closing it, and checking the conf with gedit (WTH are the gedit dev's smoking?? - it looks horrible now and seems to over-ride my WM and appearance settings). Geometry setting in
~/.config/geany/geany.conf
is still:
geometry=0;0;1358;713;0;
I assume this reflects my unchecked save-window-position-and-geometry:
pref_main_save_winpos=false

Geany still opens at about half this size (unless I set it to save position and geom, which works as expected).

Openbox ATM unfortunately, is still unable to set window geometry. It can set origin position (i.e., the position of the upper left corner) and it can do that either as a suggestion that can be over-ridden by the app or "forced" which I think can still be over-ridden, but not as easily. If Openbox COULD do this it would be great, but it can't. Not yet anyway.

man geany saith:
"Geany supports all generic GTK options, a list  is  available  on  the help screen."
From  the menu bar Help; Help takes me to a web page with repeats the above sentence.
From CL, "geany --help" doesn't have anything germane; neither does "geany --help-gtk". I even grepped for "geometry" just to be sure I wasn't reading right over it. I coulda sworn there was a generic gtk geometry option, but I haven't found evidence of it.

I tried as command options, all the likely seeming strings like:
geany -geometry=1300x500+0+0
geany --geometry=1760x750+0+0
geany --geometry 1760x750+0+0
geany -geometry 1760x750+0+0
geany -geometry=1760x750+0+0
geany --geometry=1760x750

but they all just return "Geany: Unknown option . . .".

I suppose I can wrap geany in a script  that zaps the window with wmctrl, but I keep thinking there ought to be a more elegant way to give it a default geometry. Where could the default it seems to act on be coming from? It is the same half-screen size every time, so it must be getting those dimensions from somewhere.

**Comment by elextr** [comment]
The GTK2 docs don't actually mention [--geometry](https://developer.gnome.org/gtk2/2.24/gtk-running.html) (nor in the page on [X11 options](https://developer.gnome.org/gtk2/2.24/gtk-x11.html)) so I guess thats why it doesn't work.

The position of "Save window position and geometry" in the "Shutdown" section of the dialog is slightly misleading, it not only stops Geany saving it, it also stops it using any value in the file on startup and you will get whatever your WM decides.

**Comment by Lew-Rockwell-Fan** [comment]
> The position of "Save window position and geometry" in the "Shutdown" section of the dialog is slightly misleading, it not only stops Geany saving it, it also stops it using any value in the file on startup . . .

Ahhh. Gotcha. Thanks. That's admirably clear, though I had to think about that before I fully saw the implications. So you're saying geany.conf cannot be used to set a default geometry, period. The line that looks like it would do so is, in fact, merely keeping a record of what the geometry WAS the last time geany was closed if the "remember" option is checked, and if the "remember" option is NOT checked, that line does nothing at all. There is therefore no point whatsoever in hand editing it.

If that is how it is intended to work, I suggest at least changing the name of the variable from "geometry" to "saved_geometry", or even to add a comment explaining this in the default geany.conf. Better still would be to replace "geometry" with 2 variables: "saved_geometry" and "default_geometry". So if the present behavior is intended, this is a feature request, or if it is NOT intended, this is a bug report.

> . . . and you will get whatever your WM decides.
Ahhh again. Could be. If Openbox is deciding what is, in effect, geany's default geometry then it is doing so rather badly. The failure of Openbox to provide user control of default window sizes (unless you count maximized and iconified),  through the rc.xml, has annoyed me before, but usually there is an alternate mechanism either with a config file for the the app or an option in invoking it, or both.

So, if you are confident that the effective window size is coming from OB, it looks like my options are:
1. Wait for Openbox to enhance rc.xml with window size control
2. Wait for Geany to enhance geany.conf with a default_geometry variable
3. Wrap Geany in a script and hit it with wmctrl or something similar
4. Try another WM

Is that a fair summary?
